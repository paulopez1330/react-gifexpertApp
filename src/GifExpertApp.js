import React, { useState } from 'react';
import AddCategory from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

const GifExpertApp = ({ defaultCategories =[] }) => {

    const [categories, setCategories] = useState( defaultCategories );
/*
    const hadleAdd = () => {
        setCategories( [...categories, 'Hounter'] );
    }
  */
    return ( 
        <>
            <h2>GifExpertApp</h2>
            <AddCategory setCategories = { setCategories } />
            <hr />
            <ol>
                {
                    categories.map( cat => (
                        <GifGrid
                            key={ cat }
                            category={ cat }
                        /> 
                    ))
                }
            </ol>
        </>
     );
}

export default GifExpertApp;