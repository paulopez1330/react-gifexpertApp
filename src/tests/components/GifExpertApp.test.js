import React from 'react';
import '@testing-library/jest-dom';

import { shallow } from 'enzyme';
import GifExpertApp from '../../GifExpertApp';

describe('Pruebas en <GifExpertApp/>', () => {
    

    beforeEach( () => {
        
        jest.clearAllMocks();

    });

    test('debe de mostrar componente correctamente', () => {
                
        const wrapper = shallow( <GifExpertApp /> )
        expect( wrapper ).toMatchSnapshot();
    });

    test('debe de mostrar la lista de categorias', () => {
        const categories = [ 'One Punch', 'Dragon Ball' ];
        
        const wrapper = shallow( <GifExpertApp defaultCategories = { categories } /> )
        
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find( 'GifGrid' ).length ).toBe( categories.length );

    });

})