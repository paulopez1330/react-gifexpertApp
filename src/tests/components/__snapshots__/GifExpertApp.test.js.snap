// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp/> debe de mostrar componente correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Pruebas en <GifExpertApp/> debe de mostrar la lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;
